### PR TITLE
Split every class/enum/struct into it's own file

### PR DESCRIPTION
**Pull request by vbfox**
I created a small F# script using roslyn to extract each to it's own file.

### CODE REVIEW

**Comment by vbfox** [comment]
The code of the F# extractor is there if needed : https://gist.github.com/vbfox/f29e2649d07f0cedf6e3

**Comment by jmelosegui** [comment]
neat

File: src/Kernel32.Desktop/Kernel32+FindFirstFileExFlags.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        /// <summary>
+        /// Optional flags to pass to the <see cref="FindFirstFileEx"/> method.
+        /// </summary>
+        [Flags]
+        public enum FindFirstFileExFlags
+        {
+            /// <summary>
+            /// Searches are case-sensitive.
+            /// </summary>
+            CaseSensitive,
```
Comment (jmelosegui) [inline src/Kernel32.Desktop/Kernel32+FindFirstFileExFlags.cs:22]: I think the default value here is 0,but based on [msdn](https://msdn.microsoft.com/en-us/library/windows/desktop/aa364419.aspx) `FindFirstFileExFlags.CaseSensitive` should be 1

File: src/AdvApi32.Tests/AdvApi32.cs [hunk 2]
```diff
@@ -1,10 +1,7 @@
 ﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 
-using System;
-using PInvoke;
 using Xunit;
-using static PInvoke.AdvApi32;
```
Comment (AArnott) [inline src/AdvApi32.Tests/AdvApi32.cs:7]: Removal of these using statements is a disservice, IMO. Once real tests are written, these will inevitably have to be put back. I'm guessing your tool did a 'remove unused usings' step. Can we remove that step and have the PR just extract nested types and reserve removing unused usings as a separate PR, if we can agree the result is desirable?

File: src/Kernel32.Desktop/Kernel32+CreateFileFlags.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            /// </summary>
+            IdentificationSecurity = SecurityImpersonationLevel.SecurityIdentification << 16,
+
+            /// <summary>
+            /// Impersonate a client at the impersonation level. This is the default behavior if no other flags are specified along with the SECURITY_SQOS_PRESENT flag.
+            /// </summary>
+            ImpersonationSecurity = SecurityImpersonationLevel.SecurityImpersonation << 16,
+        }
+    }
+}
```
Comment (AArnott) [inline src/Kernel32.Desktop/Kernel32+CreateFileFlags.cs]: How easy would it be to include a newline character at the end of the file to keep some tools happy (and remove the red warning indicator GitHub adds)?

**Comment by AArnott** [comment]
I left a couple of comments. 
I'd love to get this merged very soon as merge conflicts will inevitably result the longer this is held open as it touches a lot of code.

File: src/AdvApi32.Tests/AdvApi32.cs (hunk 2 above)
Comment (vbfox) [inline src/AdvApi32.Tests/AdvApi32.cs:7]: I did the removal in VS on solution scope via VS analyzerd but can undo for all non-splitted files

File: src/Kernel32.Desktop/Kernel32+CreateFileFlags.cs (hunk 3 above)
Comment (vbfox) [inline src/Kernel32.Desktop/Kernel32+CreateFileFlags.cs]: Very easy, the default roslyn Formatter doesn't do it but it's just a trivia to add

File: src/Kernel32.Desktop/Kernel32+FindFirstFileExFlags.cs (hunk 1 above)
Comment (vbfox) [inline src/Kernel32.Desktop/Kernel32+FindFirstFileExFlags.cs:22]: Yes it look like a bug, can you add an issue ?

**Comment by vbfox** [comment]
> I'd love to get this merged very soon as merge conflicts will inevitably result the longer this is held open as it touches a lot of code.

That's sure to create conflicts yes, i'll tweak the tool and reapply it

**Comment by vbfox** [comment]
I reapplied the tool on the original commit, with this time around :
- Less using cleanup, I still let some pass where it seemed waranted
- Newline at end of file
- SafeServiceHandler -> SafeServiceHandle rename (Found out as the tool was generating an empty file and I wondered why)

Code for the tool is now in https://github.com/vbfox/ExtractNestedClassesToSeparateFiles

File: src/Kernel32.Desktop/Kernel32+FindFirstFileExFlags.cs (hunk 1 above)
Comment (vbfox) [inline src/Kernel32.Desktop/Kernel32+FindFirstFileExFlags.cs:22]: Created #28

**Pull request merged**